sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1116,6 +1117,262 @@ def max_deflection(self):
+        - 6.51041666666667e-5*SingularityFunction(x, 8, 3) - 5.20833333333333e-6*SingularityFunction(x, 8, 4)
+        for x over (0.0, 8.0)
+        """
+        deflection = self.deflection()
+        for sym in deflection.atoms(Symbol):
+            if sym == self.variable:
+                continue
+            if sym not in var:
+                raise ValueError('Value of %s was not passed.' %sym)
+        if self.length in var:
+            length = var[self.length]
+        else:
+            length = self.length
+        return plot(deflection.subs(var), (self.variable, 0, length), title='Deflection',
+                xlabel='position', ylabel='Value', line_color='r')
+
+    def plot_all(self, var={}):
+        """
+        Returns a combined plot of Shear Force, Bending Moment and Deflection
+        of the Beam object.
+
+        Parameters
+        ==========
+        var : dictionary
+            Python dictionary containing Symbols as key and their
+            corresponding values.
+
+        Examples
+        ========
+        There is a beam of length 8 meters. A constant distributed load of 10 KN/m
+        is applied from half of the beam till the end. There are two simple supports
+        below the beam, one at the starting point and another at the ending point
+        of the beam. A pointload of magnitude 5 KN is also applied from top of the
+        beam, at a distance of 4 meters from the starting point.
+        Take E = 200 GPa and I = 400*(10**-6) meter**4.
+
+        Using the sign convention of downwards forces being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> R1, R2 = symbols('R1, R2')
+        >>> b = Beam(8, 200*(10**9), 400*(10**-6))
+        >>> b.apply_load(5000, 2, -1)
+        >>> b.apply_load(R1, 0, -1)
+        >>> b.apply_load(R2, 8, -1)
+        >>> b.apply_load(10000, 4, 0, end=8)
+        >>> b.bc_deflection = [(0, 0), (8, 0)]
+        >>> b.solve_for_reaction_loads(R1, R2)
+        >>> b.plot_all()
+        """
+        shear_force = self.shear_force()
+        deflection = self.deflection()
+        bending_moment = self.bending_moment()
+        for sym in deflection.atoms(Symbol):
+            if sym == self.variable:
+                continue
+            if sym not in var:
+                raise ValueError('Value of %s was not passed.' %sym)
+        if self.length in var:
+            length = var[self.length]
+        else:
+            length = self.length
+        p = plot(shear_force.subs(var), bending_moment.subs(var), deflection.subs(var),
+                (self.variable, 0, length), show=False)
+        labels = ['Shear Force', 'Bending Moment', 'Deflection']
+        colors = ['g', 'b', 'r']
+        for n, line in enumerate(p):
+            line.label= labels[n]
+            line.line_color = colors[n]
+        p.legend = True
+        return p.show()
```

Comment:
Always return the matplotlib axes and do not call show. It is convention to let the user call show.